proxy_connect.go

```diff
@@ -28,49 +29,36 @@ type BufferSource interface {
 //
 // bufferSource: specifies a BufferSource, leave nil to use default
 //
+// okWaitsForUpstream: specifies whether or not to wait on dialing upstream
+// before responding OK
+//
 // dial: the function that's used to dial upstream
 func CONNECT(
 	idleTimeout time.Duration,
 	bufferSource BufferSource,
+	okWaitsForUpstream bool,
```

Comment:
I just made this an option to the CONNECT method. The fact that Lantern happens to use false on the client and true on the server is really lantern-specific, whereas this proxy package is really intended to be more general.